android/app-example/src/main/java/com/badoo/ribs/example/rib/small/SmallInteractor.kt

```diff
@@ -31,7 +31,8 @@ class SmallInteractor(
 
     private val viewEventConsumer: Consumer<Event> = Consumer {
         when (it) {
-            Event.OpenBigClicked -> portal.showRemote(router, FullScreen.ShowBig)
+            Event.OpenBigClicked -> portal.showContent(router, FullScreen.ShowBig)
```

Comment:
Random thought: what if we make the signature of show/content type dependent?
e.g

```
fun <T : Parcelable> showContent(router: Router<in T, *, *, *>, configuration: T)
fun <T : Parcelable> showOverlay(router: Router<*,in T, *, *>, configuration: T)
```

Not sure if it will work, but additional type safety will be rather neat

Reply:
I like the idea. Unfortunately in practice the above doesn't add any kind of safety.